### ISSUE TITLE
Vampire Books - Improve way to get them

### ISSUE THREAD

**Issue by Twisted-Code** [TeamLapen/Vampirism#194]
## Versions
- Minecraft:                    1.10
- Forge:                       2185
- SPONGEFORGE:        spongeforge-1.10.2-2281-5.2.0-BETA-2353.jar
- Vampirism:                     Vampirism: 1.1.2, i think. Copy pasted from my previous issue. Probably not too important anyway since this is a suggestion

## Issue Description
I'm not sure if I'm just getting really really bad luck with the Random Number Gnome or if it's a bug, but I can't seem to get Vampire Books to drop from advanced hunters/vampires. I'm pretty sure I've killed over 20 of them by now, but alas, no book. In my opinion, even if it really is just bad luck, you should really have a less RNG-dependent  way of acquiring the books, or else remove the requirement of them for progressing. Maybe you could add some recipe or ritual involving pure blood and fangs/hearts, the latter possibly being restricted based on which faction you are? I don't know, but regardless of faction you can't even get to level 5 without them. Given I'm running a server, I need a renewable way to get them, or I'm going to have a lot of unhappy players.

I am NOT having the same issue in singleplayer, as proven by my test world a week or 2 ago when I inadvertently killed an advanced mob and got one

## Reproduce Steps
N/A

**Comment by maxanier** [comment]
Currently there is a 5% (1/20) chance of getting a book from them. So you might have to kill a few more, maybe you are just having very bad luck. 
If they are dropping other items, I can see no way Sponge might be interfering.

I understand that the current ways of getting these books (dungeon advanced mobs) is not the best. I'm planing to add a few more places to find them in 1.11+. Maybe I can think of renewable way as well.
Main intention of adding these books was to slow down the level up. I don't want players to become a top level Vampire after just a few days of playing. They should explore the world (while doing other stuff/mods) and sometimes find the books.

**Comment by Twisted-Code** [comment]
I should also add that, despite a significant amount of time put into the server and an effort to kill every advanced mob I encounter, the only vampire books I've ever obtained were from the underground vampire shrines

**Comment by Twisted-Code** [comment]
@maxanier In that case, I'll grab a stack of spawn eggs and test with 64 advanced mobs. I'll let you know whether it works out or not. I'd say if I go through 64 without a single book, something's broken, wouldn't you agree?

on the topic of renewability:
Personally, I think a good source might be, as I mentioned, some combination of pure blood and other "leveling ingredients". Perhaps the different titles of vampire books could even come to represent a different "level" of book, which might correspond to the level of pure blood used. You could add a ritual of infusion ("because every mod needs an infusion alter" ~Actually Additions' flavor text on the Empowerer) that allows infusing a book (or perhaps book & quill?) with the ingredients. Said ingredients could include liquid blood piped into the relevant altar (or, if and only if the mod cannot detect any mods that add fluid pipes, perhaps enable a blood container nearby to suffice)...  but of course, this is just my opinion. It's your mod so your call 😃.

**Comment by Twisted-Code** [comment]
Okay so it appears I'm just having very bad luck. I spawned a stack-worth of Advanced Vampire Hunter eggs, gave myself some splash potions of poison and harming, and got 3 books, and I didn't even end up killing all of them. Looks like the Random Number Gnome just hates me. Keeps stealing all my lucky numbers 😦

**Comment by Twisted-Code** [comment]
unless... do I have to be killing advanced mobs of the opposite faction? Because I've been killing advanced vampires as a vampire. I don't understand why you would make that distinction, since mob drops are mob drops, but it's worth considering, if only for that possibility to be immediately eliminated

**Comment by maxanier** [comment]
There is no distinction between factions in this case. Looks like it's all the Random Number Gnome ;)

As I said, I will try to find a better way for getting advanced books in the future.

**Comment by Twisted-Code** [comment]
yeah I'm getting books now, and fairly easily too. Side note: the Looting enchantment should increase the chance of books and DECREASE the chance of other drops (especially blood bottles) by slightly less than the increase. When farming advanced vampires using my looting sword, I end up with so many blood bottles that I don't know what to do with all this food. you might also consider giving the "luck" status effect a use by checking for it when determining mob drops, because vanilla only uses it for fishing, and who the hell bothers with fishing?

**Comment by user3287** [comment]
(Related)[Question]
Does Looting affect the drop rate?

**Comment by maxanier** [comment]
It currently does not. I will change it so, the 5% base change is increased by 1% per looting level.
Advanced mob drops should still be only a backup solution for vampire books if you just cannot find them in the world

**Comment by takashi1kun** [comment]
What about getting the books by special quests that give you a map to a special dungeon?
Ex:
Go to faction representative, do some quests, special quest appears, the reward a "Enemy base Intel Map"
When you accept it, the mod scan for some non generated chunks 1000 blocks or more away from player, then the mod generate a special dungeon structure, similar to the ones in the twilight forest dungeons around that chunk
The player then receives a map pointing to that dungeon.
Player goes to the dungeon, dungeon type depends on faction, it could be a vampire castle or a hunter's fort.
The dungeon has final boss, final boss drops book/gated item for progression
The dungeon is special, you cannot destroy blocks, after killing boss blocks became vulnerable. 
Only receive 1 map per level max, from level 4.
Boss more difficult depending on level.

Something like that, between doing quests until the special quest appears, completing it, going to the dungeon, doing the dungeon, beating the boss, returning home and that for a single book, it would take time so it couldn't really be rushed that much, but would be kind of less tedious and quite fun.

**Comment by tech-boy101** [comment]
i think better way do some simlier to how werewolf addon for vampirism dose that be deal damage to enemy is that take some time still but feel way better is you can do in is you do other things also the vampire heart one you need so many of and 2 the hunter take very long time to respawn once kill few i dont no if that so to be thing or maybe just mob cap is cause it i love know that but the heart are also pain to get and need over stack of them with take forever and not fun to do ever time you wanna become max level vampire if dont think the way werewolf mod is useable at not even increase how damage you need do what about use the age idea from on other function that was talk about be add for level up process instead so take few minecraft day be can level up get longer with each level up it also go nice with age item in first and is people all ready be use to it and even have some tell your age and all that also other wood simlier to werewolf damage bar thing but actual drink blood is you need drink x amount of blood be for can level up and make where actual need restore blood vs just drink blood even though your all full ether way i feel like curret leveling process is way to grind and not fun to do is stand right now and need decrease a lot or made easy or some you can do in back vs focus on it and it alone is vampire book take way to long to get and hearts can take just is long if not longer at time
